components/db/events.go

```diff
@@ -0,0 +1,342 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/sqlite"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+const (
+	// Event timestamp in unix seconds.
+	ColumnTimestamp = "timestamp"
+
+	// event name
+	// e.g., "memory_oom", "memory_oom_kill_constraint", "memory_oom_cgroup", "memory_edac_correctable_errors".
+	ColumnName = "name"
+
+	// event type
+	// e.g., "Unknown", "Info", "Warning", "Critical", "Fatal".
+	ColumnType = "type"
+
+	// event message
+	// e.g., "VFS file-max limit reached"
+	ColumnMessage = "message"
+
+	// event extra info
+	// e.g.,
+	// data source: "dmesg", "nvml", "nvidia-smi".
+	// event target: "gpu_id", "gpu_uuid".
+	// log detail: "oom_reaper: reaped process 345646 (vector), now anon-rss:0kB, file-rss:0kB, shmem-rss:0".
+	ColumnExtraInfo = "extra_info"
+
+	// event suggested actions
+	// e.g., "reboot"
+	ColumnSuggestedActions = "suggested_actions"
+)
+
+type Event struct {
+	// Event timestamp in unix seconds.
+	Timestamp int64 `json:"timestamp"`
+
+	// event name
+	// e.g., "memory_oom", "memory_oom_kill_constraint", "memory_oom_cgroup", "memory_edac_correctable_errors".
+	Name string `json:"name"`
+
+	// event type
+	// e.g., "Unknown", "Info", "Warning", "Critical", "Fatal".
+	Type string `json:"type"`
+
+	// event message
+	// e.g., "VFS file-max limit reached"
+	Message string `json:"message"`
+
+	// event extra info
+	// e.g.,
+	// data source: "dmesg", "nvml", "nvidia-smi".
+	// event target: "gpu_id", "gpu_uuid".
+	// log detail: "oom_reaper: reaped process 345646 (vector), now anon-rss:0kB, file-rss:0kB, shmem-rss:0".
+	ExtraInfo string `json:"extra_info"`
+
+	// event suggested actions
+	// e.g., "reboot"
+	SuggestedActions string `json:"suggested_actions"`
```

Comment:
why do we need the SuggestedActions here?

Reply:
Directly maps to our event object 

https://github.com/leptonai/gpud/blob/789b4fa4aece34f67ca9a1a2eb12c87daa3e9362/components/components.go#L79-L86

/cc @cardyok

Reply:
is this calculated based on other fields?

Reply:
not exactly..for xid, yes. but for other components might not..